### PR TITLE
Syncing - Internal repo changes.

### PR DESCRIPTION
**Pull request by pmzara**
*Issue #, if available:*

*Description of changes:*
- Revert "Added support in the agent for Ubuntu LTS release("bionic") ships with Ruby 2.5.x." 
- Remove procfs dependency to make agent more compatible with other operating systems
- Use S3 Fips endpoints 
- Add use_fips_mode option 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/instance_agent/plugins/codedeploy/command_executor.rb [hunk 1]
```diff
@@ -238,33 +238,8 @@ def most_recent_install_file_path(deployment_group)
         private
         def download_from_s3(deployment_spec, bucket, key, version, etag)
           log(:debug, "Downloading artifact bundle from bucket '#{bucket}' and key '#{key}', version '#{version}', etag '#{etag}'")
-          region = ENV['AWS_REGION'] || InstanceMetadata.region
```
[the remark this answers is not in the document]
Comment (pmzara) [reply lib/instance_agent/plugins/codedeploy/command_executor.rb]: The code wasn't removed. It was only moved to the new function s3_options to become more clear.

[the remark this answers is not in the document]
Comment (annamataws) [reply lib/instance_agent/plugins/codedeploy/command_executor.rb]: You have
FIPS_ENABLED_REGIONS = Set['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'us-gov-west-1', 'us-gov-east-1']  
above

Why are only 4 of those regions being special cased here?

File: test/instance_agent/config_test.rb [hunk 2]
```diff
@@ -65,5 +70,39 @@ class InstanceAgentConfigTest < InstanceAgentTestCase
         assert InstanceAgent::Config.validate_config.empty?, InstanceAgent::Config.validate_config.inspect
       end
     end
+    
+    context 'validate use_fips_mode' do
```
[the remark this answers is not in the document]
Comment (annamataws) [reply test/instance_agent/config_test.rb]: You have
FIPS_ENABLED_REGIONS = Set['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'us-gov-west-1', 'us-gov-east-1']
above

Why are only 4 of those regions being special cased here?

[the remark this answers is not in the document]
Comment (annamataws) [reply test/instance_agent/config_test.rb]: Talked with Peng

**Pull request merged**